### DISCUSSION TITLE
Anybody deployed librechat on AWS ECS

### DISCUSSION THREAD

**Discussion by saeed-siddiqui** [danny-avila/LibreChat#4845] [Q&A]
Trying to figure out the best option to deploy Librechat. EC2 or AWS ECS?

**Comment by gshockbu** [comment]
We are planning on an ECS Fargate deployment using containers.

**Reply by madrover** [reply]
Nice!
Looking forward to hear more about this

**Reply by markusmundt** [reply]
We use ecs fargate for librechat and it runs fine. Only thing is the librechat.yaml cause we couldn‘t bind it like the .env via s3 bucket. Any ideas for this issue? Thank you!!!

**Reply by sudheermanubolu** [reply]
hi, you can have a init container with aws-cli image to copy librechat.yaml from S3 to your task container. Below is a sample cdk to deploy librechat on AWS.
https://github.com/sudheermanubolu/librechat-cdk

**Reply by markusmundt** [reply]
Thank you!! I found a way that works for our environment.

**Comment by sudheermanubolu** [comment]
ECS is better as it can scale better by separating the services (librechat, meilisearch & rag) into different fargate instances. If you have existing k8s cluster or experience, k8s/EKS is also a better option.
